packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2355,14 +2401,14 @@ export class MergeTree {
 			}
 		}
 
-		const newOrder = Array.from(affectedSegments.map(({ data }) => data));
-		newOrder.forEach((seg) =>
-			seg.localRefs?.walkReferences((lref) => lref.callbacks?.beforeSlide?.(lref)),
-		);
+		// eslint-disable-next-line unicorn/no-useless-spread
+		const newOrder = [...affectedSegments.map(({ data }) => data)];
```

Comment:
`.map` returns an array, doesn't it? Is this extra copy needed?

Reply:
affectedSegments isn't an array, so no.

Similar to Tim's other comment, if you can get away with `Array.from(affectedSegments, ({ data }) => data );` that seems fine

Reply:
that matches the style higher up in the function as well and doesn't seem like that one caused any linter complaints, so I think it should be ok